### PR TITLE
feat: implement `IN` operator

### PR DESCRIPTION
**Pull request by khvn26**

### CODE REVIEW

File: Flagsmith.Engine/Segment/Evaluator.cs [hunk 1]
```diff
@@ -53,16 +53,16 @@ static bool TraitsMatchSegmentCondition(List<TraitModel> identityTraits, Segment
 
         public static bool MatchesTraitValue(object traitValue, SegmentConditionModel condition)
         {
-            var exceptionOperatorMethods = new Dictionary<string, string>()
-            {
-                { Constants.NotContains, "EvaluateNotContains" },
-                { Constants.Regex, "EvaluateRegex" },
-                { Constants.Modulo, "EvaluateModulo" },
```
Comment (matthewelwell) [inline Flagsmith.Engine/Segment/Evaluator.cs:60]: I'd rather revert the formatting change here if possible. Not 100% sure if this is correct formatting for c# but it's currently consistent throughout the codebase so we should try to keep it that way (or submit a PR to change it everywhere if we think it's wrong).

**Review by matthewelwell — CHANGES_REQUESTED** [review]

File: Flagsmith.Engine/Segment/Evaluator.cs [hunk 2]
```diff
@@ -53,16 +53,16 @@ static bool TraitsMatchSegmentCondition(List<TraitModel> identityTraits, Segment
...
+                {Constants.In, "EvaluateIn"},
+                {Constants.NotContains, "EvaluateNotContains"},
+                {Constants.Regex, "EvaluateRegex"},
+                {Constants.Modulo, "EvaluateModulo"},
             };
 
             if (exceptionOperatorMethods.TryGetValue(condition.Operator, out var operatorMethod))
             {
-                return (bool)typeof(SegmentConditionModel).GetMethod(operatorMethod).Invoke(condition, new object[] { traitValue });
+                return (bool)typeof(SegmentConditionModel).GetMethod(operatorMethod).Invoke(condition, new object[] { traitValue.ToString() });
```
[the remark this answers is not in the document]
Comment (khvn26) [reply Flagsmith.Engine/Segment/Evaluator.cs:66]: > Based on this, we should probably change the signature of the EvaluateModulo method?

Done.

> Perhaps we should also make it obvious that we're casting the traitValue to double (pull it to the top of the method), but maybe that's less important?

I think it's pretty obvious as it is.

File: Flagsmith.Engine/Segment/Evaluator.cs (hunk 1 above)
Comment (khvn26) [reply Flagsmith.Engine/Segment/Evaluator.cs]: Sure, done.
Any tips for setting up auto-formatting to support the common style?

Comment (matthewelwell) [reply Flagsmith.Engine/Segment/Evaluator.cs]: Sorry, forgot to respond here. I don't have anything built in actually. I just run `dotnet format` manually I think... we should add this to a pre-commit hook.

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**